### DISCUSSION TITLE
resource-query not built by default?

### DISCUSSION THREAD

**Discussion by wkharold** [flux-framework/flux-sched#1135] [Q&A] [answered]
It looks like a simple 

```make; make install```

 of `flux-sched` doesn't build, or install, the `resource-query` utility. What is the best way to make it available?

**Comment by garlick** [comment]
The sched devs should answer this but knowing the use case may be helpful.  It appears to be an internal test-only tool at this point.

Note: this was brought up before here:  https://github.com/flux-framework/flux-sched/discussions/869 (and not answered).

Edit: it does build for me, to `$(builddir)/resource/utilities/resource-query`

**Reply by wkharold** [reply]
Right, it seems like it should get built. I need to look at the logs and see if there were any errors.

**Reply by wkharold** [reply]
OK, so the first problem was that cmake wasn't installed. Now 

```bash
./configure --prefix=/usr/local
```

gives me:

```
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Python (missing: Python_INCLUDE_DIRS Python_LIBRARIES
  Development Development.Module Development.Embed) (found suitable version
  "3.11.5", minimum required is "3.6")
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindPython/Support.cmake:3766 (find_package_handle_standard_args)
  /usr/share/cmake/Modules/FindPython.cmake:580 (include)
  CMakeLists.txt:67 (find_package)
```

**Reply by grondo** [reply, marked as answer]
That indicates the development package (headers and libs) is not installed for that version of Python

**Reply by wkharold** [reply]
Quick follow-up. I'm on Rocky Linux 8 where the system Python is 3.6. I install several 3.6 packages  (devel, cffi, yaml, jsonschema, sphinx, docutils) to get flux-security (and maybe flux-core) to build and install. It seems like flux-core is installing Python 3.11 so I also install that, along with devel and pip, in the base image and then pip install 3.11 cffi, pyyaml, jsonschema, and sphinx. 

All of this yields a working flux system instance on Google Cloud but it seems kind of hacky. Am I missing something (very probably) or is this a viable approach?

**Reply by grondo** [reply]
Multiple versions of Python should not be a requirement. Our test image may install Python 3.11, but I'm guessing this is just for testing in CI.

The requirements to build and test flux-core are contained in [scripts/install-deps-rpm.sh](https://github.com/flux-framework/flux-core/blob/master/scripts/install-deps-rpm.sh). For Fluxion you will need any additional dependencies listed in the [README](https://github.com/flux-framework/flux-sched#installing-redhatcentos-packages).

Otherwise, your approach seems sound. Note that if you want Python bindings built for a different Python version than the default, you will need to build them separately (and I think we have a couple different methods if you need that!)
